pytorch_lightning/overrides/fairscale.py

```diff
@@ -0,0 +1,108 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Dict, Any, List, Tuple, Union
+
+from fairscale.nn.data_parallel.sharded_ddp import ShardedDataParallel
+from fairscale.optim import OSS
+from torch import nn
+
+from pytorch_lightning.utilities import rank_zero_warn
+
+
+class LightningOSS(OSS):
+
+    def state_dict(self) -> Dict[str, Any]:
```

Comment:
n00b question: why do we override the state dict? how do we ensure this stays in sync with upstream fairscale?